### PR TITLE
ListOptions: add test for ResourceVersion > 0 in List

### PR DESCRIPTION
**Pull request by hongchaodeng**
ref: #28472 

Done:
- Add a test for ResourceVersion > 0 in registry (cache) store List()
- Fix the docs.

### CODE REVIEW

**Comment by ixdy** [comment]
@k8s-bot test this please, issue #27462

**Comment by hongchaodeng** [comment]
@wojtek-t 
Ready for review.
Can you take over and help review it? Thanks!

File: pkg/api/types.go [hunk 1]
```diff
@@ -2226,7 +2226,10 @@ type ListOptions struct {
 	FieldSelector fields.Selector
 	// If true, watch for changes to this list
 	Watch bool
-	// The resource version to watch (no effect on list yet)
+	// For watch, it's the resource version to watch.
+	// For list, if unset (or zero), there is basically no guarantee on what List will return;
```
Comment (lavalamp) [inline pkg/api/types.go]: The return should include a resource version which you can still use against a watch and not miss anything. Additionally, this depends on the quorum read flag. If it is on, then this comment is wrong.

@wojtek-t

Comment (wojtek-t) [inline pkg/api/types.go]: Yes - the result will always contain resourceVersion of the returned result.

@hongchaodeng - I think I missed something when you asked me about it. So let me put this together again.
In cache the semantic is the following:
- if unset - redirect to etcd_helper, and then the result is returned from etcd based on quorum-read flag
- if 0 - then we simply return what we currently have, no guarantee
- in set to non zero - the result at least as fresh as given rv is returned

In all cases, the result contains RV from which it comes.

Can you please update this comment (sorry for confusion previously).

**Comment by hongchaodeng** [comment]
@wojtek-t @lavalamp @xiang90 
Addressed comments and tests passed.

**Comment by wojtek-t** [comment]
lgtm - thanks!

**Comment by wojtek-t** [comment]
@k8s-bot test this, issue: #IGNORE  (verification didn't finish)

**Pull request merged**